source/industry/Fanuc.rst

```diff
@@ -0,0 +1,121 @@
+  Relevant hardware                   R-30iA or R-J3iC (controller)
+  =================================   =======================================
+  
+  
+
+**Further reading**
+
+   * `manual_collection`_
+   * `manual_slides`_
+   * `roboguide_help`_
+   * `reference_manual`_
+   * `Karel`_
+
+   
+
+Trajectory composition
+----------------------
+Programming is done with move instructions (robot movement types).  (see `manual_slides`_ p. 15-16):
+
+* Linear Cartesian motions
+   **LINEAR**: controlled movement of the TCP in a straight line from position A to B
+
+* Circular motions
+   **CIRCULAR**: The TCP follows a circular arc from the initial position to the destination
+   
+* Joint space interpolation
+   **JOINT**: basic robot motion with nonlinear toolpath. Tool speed is determined with % of the maximum speed.
+
+   
+
+Waypoint representation
+-----------------------
+Points are described with position coordinates x,y, z and rotations w, p, r.
+
+.. code-block:: yaml
+
+  x
+  y
+  z
+  w (x-axis rotation)
+  p (y-axis rotation)
+  r (z-axis rotation)
+
+
+Trajectory parameterization and execution
+-----------------------------------------
+
+(see `reference_manual`_)
+
+Specification of velocity
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+System variables determine the speed of robot motion 
+
+* Speed overrides:
+
+    * `$MCR.$GENOVERRIDE`: general override
+    * `$MCR_GRP.$PROGOVERRIDE`: program override
+
+* Manual Motion Speed
+
+    * Joint Speed (in joint units) equals: `$PARAM_GROUP[i].$SPEEDLIMJNT` * `$MCR[].$GENOVERRIDE` / 100  * `$SCR_GRP[i].$JOGLIM_JNT` / 100
+    * Cartesian Translational Speed (in mm/sec) equals: $PARAM_GROUP[i].SPEEDLIM * $MCR[].$GENOVERRIDE / 100 * $SCR_GRP[i].$JOGLIM / 100
+    * Cartesian Rotational Speed (in mm/sec) equals: ($PARAM_GROUP[i].ROTSPEEDLIM * $MCR[].$GENOVERRIDE / 100 * $SCR_GRP[i].$JOGLIMROT / 100
+
+* Programmed Motion Speed
+
+
+specification of acceleration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Specification of acceleration can be done via the following variables:
+
+* acceleration time is fixe and proportional to the programmed speed.
+* **$USEMAXACCEL**: enables 'fast acceleration' feature
+
+
+Blending
+~~~~~~~~
+
+Taught positions can either be fly-by points, or stop points:
+
+* **FINE**: motion stops robot arm briefly at each way point
+* **CNT**: robot approaches to the point with a distance specified by the CNT value without ever actually reaching the point.
+
+
+Parallel IO operations
+~~~~~~~~~~~~~~~~~~~~~~
+
+No information found so far
+
+Online (real-time) trajectory modifications
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Dynamic Path Modifier (DPM)
+
+* dynamic path modification using sensor data, so robot's path can be adapted in real-time
+* an external sensor provides postion and orientation offset for the next destination
+* applicable to multiple groups
+* possible applications:
+
+    * Weave operations
+    * Stationary tracking
+    * Orientation control
```

Comment:
The `J519` option (or *Stream Motion*) is missing here.

It's similar to KUKA RSI and ABB EGM.

Unfortunately it's only available in the US.

Reply:
That is very interesting. However, I could not find much information about R912.
```suggestion
    * Orientation control
    
J519  (Stream Motion)     
    * external protocol for:
        * path trajectory planning
        * near-real time streaming of the path trajectory to the robot 
        * enabling highly flexible and dynamic applications
    *  availability is restricted to the US
    
R912 (Remote Motion Interface)       
    * drip-feed for TP programs
    
```

Reply:
>  I could not find much information about R912.

It's a relatively new option. That's probably the reason.

Note that the restriction to the US of J519 is only something I've heard about. It's not official policy necessarily.

Reply:
ah thanks. I therefor decided to neglect the information about the availability